### DISCUSSION TITLE
onSuccess behavior on windows run before tsup finish

### DISCUSSION THREAD

**Discussion by axmad386** [egoist/tsup#1057] [Q&A]
I don't know how describe my problem correctly. When I use tsup in watch mode, then use `onSuccess` to trigger something, it's works on linux but on windows the behavior is strange. This is the script I run
```
tsup --watch --onSuccess 'node dist/main.js'
```
On linux, the behaviour is correct, but on windows, `node dist/main.js` run before tsup finishing bundling so it will throw error that `dist/main.js` doesn't exist.  Did I miss something?

**Comment by maxmueh** [comment]
still happening
